siliconcompiler/schema/schema_cfg.py

```diff
@@ -3848,6 +3849,40 @@ def schema_constraint(cfg):
             may adjust sizes to meet competing goals such as manufacturing design
             rules and grid placement guidelines.""")
 
+    metrics = {'width': ['width', 1.0],
+               'length': ['length', 1.0],
+               'height': ['height', 1.0]
+               }
+
+    for i, v in metrics.items():
+        scparam(cfg, ['constraint', 'pin', name, i],
+                unit='um',
+                sctype='float',
+                shorthelp=f"Constraint: pin {i}",
+                switch=f"-constraint_pin_{i} 'name <float>'",
+                example=[
+                    f"cli: -constraint_pin_{i} 'nreset {v[1]}'",
+                    f"api: chip.set('constraint', 'pin', 'nreset', {i}, {v[1]})"],
+                schelp=f"""
+                Pin {i} constraint. The parameter is a goal/intent, not an exact
+                specification. The layout system may adjust sizes to meet
+                competing goals such as manufacturing design rules and grid placement
+                guidelines.""")
+
+    scparam(cfg, ['constraint', 'pin', name, 'shape'],
+            sctype='enum',
+            enum=['circle', 'rectangle', 'square',
+                  'hexagon', 'octagon', 'oval', 'pill', 'polygon'],
```

Comment:
whats the difference between oval and pill?

Reply:
fixed